app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -32,15 +31,14 @@ public static boolean isTv(final Context context) {
         final PackageManager pm = App.getApp().getPackageManager();
 
         // from doc: https://developer.android.com/training/tv/start/hardware.html#runtime-check
-        boolean isTv = ((UiModeManager) context.getSystemService(UI_MODE_SERVICE))
+        boolean isTv = ContextCompat.getSystemService(context, UiModeManager.class)
                 .getCurrentModeType() == Configuration.UI_MODE_TYPE_TELEVISION
                 || pm.hasSystemFeature(AMAZON_FEATURE_FIRE_TV)
                 || pm.hasSystemFeature(PackageManager.FEATURE_TELEVISION);
 
         // from https://stackoverflow.com/a/58932366
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            final boolean isBatteryAbsent
-                    = ((BatteryManager) context.getSystemService(BATTERY_SERVICE))
+            final boolean isBatteryAbsent = context.getSystemService(BatteryManager.class)
```

Comment:
Not ContextCompat here?

Reply:
It isn't needed, as `Context`'s `getSystemService(Class)` method is available on Android Marshmallow and higher.

Reply:
Sorry, I don't understand. Don't you always need to call `context.getSystemService(Class)`.

Reply:
`ContextCompat.getSystemService(Context, Class)` actually calls `Context.getSystemService(Class)` on Android Marshmallow and higher, and calls `Context.getSystemService(String)` on lower versions.